crypto/evp/p_lib.c

```diff
@@ -84,14 +85,27 @@ int EVP_PKEY_save_parameters(EVP_PKEY *pkey, int mode)
     return 0;
 }
 
+static int evp_pkey_is_pure_provided(const EVP_PKEY *pk)
+{
+    return pk->ameth == NULL && pk->pkeys[0].keymgmt != NULL;
+}
+
 int EVP_PKEY_copy_parameters(EVP_PKEY *to, const EVP_PKEY *from)
 {
-    if (to->type == EVP_PKEY_NONE) {
-        if (EVP_PKEY_set_type(to, from->type) == 0)
-            return 0;
-    } else if (to->type != from->type) {
-        EVPerr(EVP_F_EVP_PKEY_COPY_PARAMETERS, EVP_R_DIFFERENT_KEY_TYPES);
-        goto err;
+    /*
+     * Only check that type match this early when both keys are legacy.
+     * If either of them is purely provided, we let evp_keymgmt_util_copy()
+     * do this check, after having exported either of them that isn't
+     * provided.
+     */
+    if (!(evp_pkey_is_pure_provided(to) && evp_pkey_is_pure_provided(from))) {
```

Comment:
This logic doesnt match the comment does it? Dont you want !X && !Y
You are currently testing is either key a legacy key?

Reply:
You're right.  I'll have another look in the morning...

Reply:
Cheers, I have just about finished reviewing.. I dont want you being held up. Is there any other PR's you need to be reviewed?

Reply:
I need #11074 in place before tackling this one